driver-core/src/main/com/mongodb/client/model/Aggregates.java

```diff
@@ -920,6 +925,91 @@ public static Bson searchMeta(final SearchCollector collector, final SearchOptio
         return new SearchStage("$searchMeta", notNull("collector", collector), notNull("options", options));
     }
 
+    /**
+     * Creates an $unset pipeline stage that removes/excludes fields from documents
+     *
+     * @param fields the fields to exclude. May use dot notation.
+     * @return the $unset pipeline stage
+     * @mongodb.driver.manual reference/operator/aggregation/project/ $unset
+     * @mongodb.server.release 4.2
+     * @since 4.8
+     */
+    public static Bson unset(final String... fields) {
+        return unset(Arrays.asList(fields));
+    }
+
+    /**
+     * Creates an $unset pipeline stage that removes/excludes fields from documents
+     *
+     * @param fields the fields to exclude. May use dot notation.
+     * @return the $unset pipeline stage
+     * @mongodb.driver.manual reference/operator/aggregation/project/ $unset
+     * @mongodb.server.release 4.2
+     * @since 4.8
+     */
+    public static Bson unset(final List<String> fields) {
+        if (fields.size() == 1) {
+            return new BsonDocument("$unset", new BsonString(fields.get(0)));
+        }
+        BsonArray array = new BsonArray();
+        fields.stream().map(BsonString::new).forEach(array::add);
+        return new BsonDocument().append("$unset", array);
+    }
+
+    /**
+     * Creates a $geoNear pipeline stage that outputs documents in order of nearest to farthest from a specified point.
+     *
+     * @param near The point for which to find the closest documents.
+     * @param distanceField The output field that contains the calculated distance.
+     *                      To specify a field within an embedded document, use dot notation.
+     * @param options {@link GeoNearOptions}
+     * @return the $geoNear pipeline stage
+     * @mongodb.driver.manual reference/operator/aggregation/project/ $geoNear
+     * @since 4.8
+     */
+    public static Bson geoNear(
+            final Point near,
+            final String distanceField,
+            final GeoNearOptions options) {
+        notNull("near", near);
+        notNull("distanceField", distanceField);
+        notNull("options", options);
+        return new Bson() {
+            @Override
+            public <TDocument> BsonDocument toBsonDocument(final Class<TDocument> documentClass, final CodecRegistry codecRegistry) {
+                Document d = new Document();
```

Comment:
Any reason not to follow the pattern in BucketStage, where it uses BsonDocumentWriter directly?

Reply:
Updated